### ISSUE TITLE
Operations within pure enum require scope qualifier

### ISSUE THREAD

**Issue by gmpreussner** [nim-lang/Nim#3417]
I would have expected the following to work:

``` nim
MyEnum {.pure.} = enum
  foo = 1,
  bar = 2
  foobar = ord(foo) or ord(bar)
```

But the compiler can't find 'foo' nor 'bar', even though we're within the enum's scope.

Workaround 1 (qualify enum scope):

``` nim
MyEnum {.pure.} = enum
  foo = 1,
  bar = 2
  foobar = ord(MyEnum.foo) or ord(MyEnum.bar)
```

Workaround 2 (make it not pure):

``` nim
MyEnum = enum
  foo = 1,
  bar = 2
  foobar = ord(foo) or ord(bar)
```

**Comment by Varriount** [comment]
Uh, isn't that the whole point of pure enumerations?

**Comment by gmpreussner** [comment]
If I'm already inside the scope, why would I have to specify it again? Seems kind of redundant, no?

**Comment by andreaferretti** [comment]
Seems fixed since at least v0.18, closing
